File: src/main/java/dev/lbuddyboy/flash/util/menu/button/NextPageButton.java
package dev.lbuddyboy.flash.util.menu.button;

import dev.lbuddyboy.flash.util.bukkit.ItemBuilder;
import dev.lbuddyboy.flash.util.menu.Button;
import dev.lbuddyboy.flash.util.menu.paged.PagedMenu;
import lombok.AllArgsConstructor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

@AllArgsConstructor
public class NextPageButton<T> extends Button {

    public PagedMenu<T> menu;
    public int slot;

    @Override
    public int getSlot() {
        return slot;
    }

    @Override
    public ItemStack getItem() {

        Material material = menu.page < menu.getMaxPages() ? Material.REDSTONE_TORCH_ON : Material.LEVER;
        String name = menu.page < menu.getMaxPages() ? "&c&lNext Page" : "&c&lNo Next Page";

        return new ItemBuilder(material).setName(name).create();
    }

    @Override
    public void action(InventoryClickEvent event) {
        if (event.getClick().isLeftClick() && menu.page < menu.getMaxPages()) {
            menu.page += 1;
            menu.openMenu((Player) event.getWhoClicked());
        }
    }
}

File: src/main/java/dev/lbuddyboy/flash/util/menu/button/NextPageButton.kt
package dev.lbuddyboy.flash.util.menu.button

import dev.lbuddyboy.flash.util.bukkit.ItemBuilder
import dev.lbuddyboy.flash.util.menu.Button
import dev.lbuddyboy.flash.util.menu.paged.PagedMenu
import lombok.AllArgsConstructor
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.inventory.ItemStack

@AllArgsConstructor
class NextPageButton<T> : Button() {
    var menu: PagedMenu<T>? = null
    override var slot = 0
    override fun getSlot(): Int {
        return slot
    }

    override fun getItem(): ItemStack? {
        val material = if (menu!!.page < menu!!.maxPages) Material.REDSTONE_TORCH_ON else Material.LEVER
        val name = if (menu!!.page < menu!!.maxPages) "&c&lNext Page" else "&c&lNo Next Page"
        return ItemBuilder(material).setName(name).create()
    }

    override fun action(event: InventoryClickEvent) {
        if (event.click.isLeftClick && menu!!.page < menu!!.maxPages) {
            menu!!.page += 1
            menu!!.openMenu((event.whoClicked as Player))
        }
    }
}